avro-codegen/src/main/java/com/linkedin/avroutil1/codegen/SpecificRecordClassGenerator.java

```diff
@@ -111,6 +176,1188 @@ protected JavaFileObject generateSpecificFixed(AvroFixedSchema fixedSchema, Spec
     return javaFile.toJavaFileObject();
   }
 
+  protected JavaFileObject generateSpecificRecord(AvroRecordSchema recordSchema, SpecificRecordGenerationConfig config)
+      throws ClassNotFoundException {
+    //public class
+    TypeSpec.Builder classBuilder = TypeSpec.classBuilder(recordSchema.getSimpleName());
+    classBuilder.addModifiers(Modifier.PUBLIC);
+
+    //extends
+    classBuilder.addSuperinterface(SpecificRecord.class);
+
+    // implements
+    classBuilder.superclass(SpecificRecordBase.class);
+
+    //add class-level doc from schema doc
+    //file-level (top of file) comment is added to the file object later
+    String doc = recordSchema.getDoc();
+    if (doc != null && !doc.isEmpty()) {
+      classBuilder.addJavadoc(doc);
+    }
+
+    if(config != null) {
+      if(config.getMinimumSupportedAvroVersion().laterThan(AvroVersion.AVRO_1_7)) {
+        // MODEL$ as SpecificData()
+        classBuilder.addField(
+            FieldSpec.builder(ClassName.get("org.apache.avro.specific", "SpecificData"), "MODEL$", Modifier.PRIVATE,
```

Comment:
ClassName.get("org.apache.avro.specific", "SpecificData") could be made a const (and others similarly)